### PR TITLE
Fix version checking for SUNDIALS

### PR DESCRIPTION
**Pull request by Steven-Roberts**
`find_package` required version SUNDIALS 4.x.x as opposed to >=4.0.0.

### CODE REVIEW

**Comment by Steven-Roberts** [comment]
It seems the variables added to `CONFIG_LDFLAGS` here are never set. Is this line needed?

https://github.com/boutproject/BOUT-dev/blob/ed69fd674bef813f01656bd99962fad880b922f9/cmake/SetupBOUTThirdParty.cmake#L306

File: cmake/SetupBOUTThirdParty.cmake [hunk 1]
```diff
@@ -297,7 +297,11 @@ if (BOUT_USE_SUNDIALS)
     FetchContent_MakeAvailable(sundials)
     message(STATUS "SUNDIALS done configuring")
   else()
-    find_package(SUNDIALS 4 REQUIRED)
+    ENABLE_LANGUAGE(C)
+    find_package(SUNDIALS REQUIRED)
```
Comment (ZedThree) [inline cmake/SetupBOUTThirdParty.cmake:301]: I take it:

```suggestion
    find_package(SUNDIALS 4.0.0 REQUIRED)
```

doesn't work?

**Comment by ZedThree** [comment]
That line is used to set the linking flags that `bout-config` outputs. Those variables were probably set in previous versions (of BOUT++). They are a bit tricky to get robustly and generically from CMake, so we probably need to revisit how to set these.

File: cmake/SetupBOUTThirdParty.cmake (hunk 1 above)
Comment (Steven-Roberts) [reply cmake/SetupBOUTThirdParty.cmake:301]: No, that only seems to work for 4.x.x. For a newer version, I get a warning like
```
CMake Warning at cmake/FindSUNDIALS.cmake:34 (find_package):
  Could not find a configuration file for package "SUNDIALS" that is
  compatible with requested version "4".

  The following configuration files were considered but not accepted:

    .../SUNDIALSConfig.cmake, version: 7.0.0

Call Stack (most recent call first):
  cmake/SetupBOUTThirdParty.cmake:301 (find_package)
  CMakeLists.txt:479 (include)
```
from [here](https://github.com/boutproject/BOUT-dev/blob/ed69fd674bef813f01656bd99962fad880b922f9/cmake/FindSUNDIALS.cmake#L34) (with QUIET removed). This leads to a linking error later.

With cmake 3.19, I think `find_package(SUNDIALS 4...7 REQUIRED)` might work.

Comment (ZedThree) [reply cmake/SetupBOUTThirdParty.cmake:301]: Oh yes, pretty sure `4...7` is the correct syntax

Comment (Steven-Roberts) [reply cmake/SetupBOUTThirdParty.cmake:301]: Turns out that the `4...7` syntax doesn't work with
https://github.com/LLNL/sundials/blob/b685654c6958699821fc68a822b66076fec8cae7/CMakeLists.txt#L267
so the current manual check is probably the best option.

Comment (ZedThree) [reply cmake/SetupBOUTThirdParty.cmake:301]: Ah, yes, I see now, that's very unfortunate. `SameMajorVersion` means "the requested version must have exactly the same major version", and makes no allowances for requesting "any of these major versions".

Agreed, the manual check is the best option.

File: cmake/SetupBOUTThirdParty.cmake [hunk 2]
```diff
@@ -297,7 +297,11 @@ if (BOUT_USE_SUNDIALS)
     FetchContent_MakeAvailable(sundials)
     message(STATUS "SUNDIALS done configuring")
   else()
-    find_package(SUNDIALS 4 REQUIRED)
+    ENABLE_LANGUAGE(C)
```
Comment (ZedThree) [inline cmake/SetupBOUTThirdParty.cmake:300]:
```suggestion
    enable_language(C)
```

**Review by ZedThree — APPROVED** [review]

**Comment by ZedThree** [comment]
Thanks @Steven-Roberts !

**Comment by ZedThree** [comment]
Numpy 2.0 is out, and we're getting some incompatible versions in CI which is causing the job failures

**Review by bendudson — APPROVED** [review]

**Pull request merged**